src/react-components/debug-panel/AudioDebugPanel.js

```diff
@@ -109,8 +113,14 @@ export function AudioDebugPanel({ isNarrow, collapsed, onCollapsed }) {
     } else {
       setAvatarSettings(getCurrentAudioSettingsForSourceType(SourceType.AVATAR_AUDIO_SOURCE));
     }
-    for (const [el, audio] of APP.audios.entries()) {
-      updateAudioSettings(el, audio);
+    if (qsTruthy("newLoader")) {
```

Comment:
Unless I'm misunderstanding things, `"newLoader"` does not mean all audios in the app are loaded with bitecs... For example, avatars will continue to be loaded through the old code paths, and avatar gltfs can have `audios` in them.

Beyond that though, I don't understand the purpose of this change. `updateAudioSettings` already handles whether the `audio` is a bitecs entity or aframe element. Why move that check outside to here?

Reply:
Yep thats right. We are already using the audio system in things like the video component, it should already support both. And yes they will need to co-exist since avatars and the pen tool will continue to be aframe entities even after we make newLoader the default

Reply:
This PR is closely related to https://github.com/mozilla/hubs/pull/5974

For components like video, we are already adding audio objects to `APP.audios` based on eid but for components that we are not yet migrating like `avatar-audio-source` we still add them by element. New bit-ecs systems like `audio-zones-system` or `audio-debug-system` rely on eids. It looked simpler to me to make `avatar-audio-source` have two paths for the new and old loaders: 
https://github.com/mozilla/hubs/blob/d6118639a6d1f2d3cd4d04e7457bc675bf783f28/src/components/avatar-audio-source.js#L86-L91
than moving them to adding audios to `APP.audios` using el.eid or adding conditional code in the new systems to check for el/eid Vs creating similar methods that operate on eid and just use `APP.world.eid2obj` to get to get the audio objects, which also sounds like a logical next step as it avoids having duplicated references in `APP.audios` and `APP.world.eid2obj`. This also simplifies the migration of  components like `scale-audio-feedback` and `morph-audio-feedback`.

This duplicated code it's just a first step toward that decoupling of the audios stores. Both old and new loaders would work the same but new loader will use in `APP.world.eid2` and eids for all audios stores and the old loader will keep on using elements.

Reply:
Please not that the links above make reference to the https://github.com/mozilla/hubs/pull/5974 PR which builds on top of this one. I recommend to take a look at that one as part of this review to have the full picture.

Reply:
In your PR summary you said,

> This PR removes dependency on `APP.audios` for the bit-ecs systems as the audio object is already stored in `APP.world.obj2eid`. More store migration will happen in next PRs as we move bit-ecs systems away from APP audio stores.

We should not make any changes to the audio stores this until after the migration. 

`APP.audios` and the rest of the audio state maps have key types of `AEntity | EntityID` because they are used by both code paths. There's no advantage to changing these now, and increases the risk that we introduce regressions. This line is an example of such a regression. When `newLoader` is enabled, this code will (incorrectly) ignore any audios that are associated with `AEntity`s. 

It will be significantly easier to change how we store audio data after we remove `AEntity`s, because then the audio state maps will be `Map<EntityID, Foo>`. Until then, please reverse the changes to how audio data is stored.

Reply:
> It looked simpler to me to make avatar-audio-source have two paths for the new and old loaders... than moving them to adding audios to APP.audios using el.eid or adding conditional code in the new systems to check for el/eid Vs creating similar methods that operate on eid and just use APP.world.eid2obj to get to get the audio objects

I don't understand this concern. To get the audio in the old (existing) code, we use `APP.audios.get`. To get the audio in the new code, we can still use `APP.audios.get`. Similarly for all of the other audio related state. That's what will allow us to migrate with very little effort.

>  it avoids having duplicated references in APP.audios and APP.world.eid2obj
This duplicate state goes away trivially once we're done with the migration.

Reply:
Answering to my own question I think we can avoid bridging the two systems by iterating `APP.adios` instead of using the `AudioEmitter` query. It will require to update the existing bit systems but I think it can work and it will avoid the code duplication for bridging until we migrate the avatar source component